grails-app/domain/org/transmartproject/db/dataquery/highdim/tworegion/DeTwoRegionJunction.groovy

```diff
@@ -0,0 +1,71 @@
+package org.transmartproject.db.dataquery.highdim.tworegion
+
+import groovy.transform.EqualsAndHashCode
+import org.transmartproject.core.dataquery.highdim.tworegion.Junction
+import org.transmartproject.db.dataquery.highdim.DeSubjectSampleMapping
+
+/**
+ * Created by j.hudecek on 4-12-2014.
+ */
+@EqualsAndHashCode()
+class DeTwoRegionJunction implements Serializable, Junction {
+
+    Long upEnd
+    String upChromosome
+    Long upPos
+    Character upStrand
+    Long downEnd
+    String downChromosome
+    Long downPos
+    Character downStrand
+    Boolean isInFrame
+    Long id
```

Comment:
The `Long id`. There's already an implicit `Long id` property. There is one difference though. Because you use `@EqualsAndHashCode`, `id` will be included in the comparison, which is generally not desired -- especially if the id is auto-generated, as the hash code of the object will change when it is saved to the database.